idunn/datasources/hove/models.py

```diff
@@ -0,0 +1,338 @@
+                    mode=mode,
+                )
+            ]
+        else:
+            steps = [
+                api.RouteStep(
+                    maneuver=api.RouteManeuver(
+                        location=inst.instruction_start_coordinate.to_position(),
+                        instruction=inst.instruction or inst.name,
+                        modifier=inst.get_api_modifier(),
+                    ),
+                    duration=inst.duration,
+                    distance=inst.length,
+                    geometry=geo,
+                    properties={"mode": self.mode.value if self.mode else "tram"},
+                    mode=mode,
+                )
+                for (inst, geo) in insts
+            ]
+
+        return api.RouteLeg(
+            duration=self.duration,
+            distance=self.geojson["properties"][0]["length"],
+            summary="todo",
+            steps=steps,
+            info=(
+                self.display_informations.as_api_transport_info()
+                if self.display_informations
+                else None
+            ),
+            **(
+                {
+                    "from": stops[0],
+                    "stops": stops[1:-1],
+                    "to": stops[-1],
+                }
+                if stops
+                else {}
+            ),
+        )
+
+
+class Journey(BaseModel):
+    arrival_date_time: datetime
+    departure_date_time: datetime
+    duration: int
+    distances: Distances
+    sections: List[Section]
+
+    @validator("arrival_date_time", "departure_date_time", pre=True)
+    def parse_date(cls, v):
+        return datetime.strptime(v, "%Y%m%dT%H%M%S")
+
+    def as_api_route(self) -> api.DirectionsRoute:
+        secs = [sec for sec in self.sections if sec.sec_type != SectionType.WAITING]
+        legs = [sec.as_api_route_leg() for sec in secs]
+
+        summary = [
+            summary
+            for sec in self.sections
+            if (summary := sec.as_api_route_summary_part()).info is not None
+        ]
+
+        geometry = {
+            "type": "FeatureCollection",
+            "features": [
+                {
+                    "type": "Feature",
+                    "properties": {
+                        "mode": leg.mode.value,
+                        **(
+                            {
+                                "lineColor": sec.display_informations.color,
+                                "lineTextColor": sec.display_informations.text_color,
+                            }
+                            if sec.display_informations
+                            else {}
+                        ),
+                    },
+                    "geometry": {
+                        "type": "MultiLineString",
+                        "coordinates": [step.geometry["coordinates"] for step in leg.steps],
+                    },
+                }
+                for sec, leg in zip(secs, legs)
+                for step in leg.steps
```

Comment:
Maybe it could be deleted or add the behavior in another pr ?

Reply:
Oh maybe this is misleading, leg is not specific to walking sections, this is actually roughly the equivalent of a section in the mapbox API. So this part of the code is responsible for constructing the lines that will be displayed on the map.

![Screenshot_2022-07-05_10-00-15](https://user-images.githubusercontent.com/1173464/177279591-249f351a-67b8-46f8-9ffd-f2b4f4eea5ff.png)

Reply:
I am not sure to understand. The `step` variable is not used in the code

Reply:
Oh yes indeed, that's a mistake as we already have this:

```python
"coordinates": [step.geometry["coordinates"] for step in leg.steps],
```

This is due to an older version of the code that I figured out could be simplified with a *MultiLineString*, but I forgot to remove line :sob: